indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -703,6 +704,10 @@ TaskStatus runHashPartitionMultiPhaseParallel(TaskToolbox toolbox) throws Except
             cardinalityRunner.getReports().values(),
             effectiveMaxRowsPerSegment
         );
+
+        // This is for potential debugging in case we suspect bad estimation of cardinalities etc,
+        LOG.debug("intervalToNumShards: %s", intervalToNumShards.toString());
+
```

Comment:
Apart from putting it in the log, do we need a metric around it too ?

Reply:
How often would this log message be hit? We rarely turn on debug logging until after an issue is seen. If its loggign here wont be too much, could we move to info level?

Reply:
A similar log message was there before and it was explicitly removed by a previous change.

Reply:
a metric here could be good, as @somu-imply  suggested

Reply:
I have instrumented the `Task`'s to make it easier to emit metrics...  we can delay this PR until [this other PR with the metrics instrumentation](https://github.com/apache/druid/pull/12488) merges and then I can easily add the metric.